docker-compose.yml

```diff
@@ -8,24 +13,35 @@ services:
       - 8080:8080
     volumes:
       - ./frontend:/data
+      - /data/node_modules
 
   backend:
     container_name: avement_webshop_backend
     build: ./backend
+    environment:
+      POSTGRES_HOST: postgres
+      POSTGRES_PORT: 5432
+      <<: *postgres-credentials
     ports:
       - 4000:4000
     volumes:
       - ./backend:/data
+      - /data/node_modules
     depends_on: 
       - postgres
-    environment:
-      API_URL: http://backend:4000/
-      DATABASE_URL: postgres://avement@postgres/
```

Comment:
Depending on the ORM you want to use, it is better to have the DB configuration separated into more granular keys like this:

POSTGRES_HOST: postgres
POSTGRES_PORT: 5432
POSTGRES_USER: admin
POSTGRES_PASSWORD: admin
POSTGRES_DB: avement

Reply:
For this, i can recommend TypeORM or MikroORM.
I would also very much recommend using TypeScript, since working with database entities without types can be quite the hassle!